llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp

```diff
@@ -2157,10 +2159,16 @@ PreservedAnalyses PGOInstrumentationUse::run(Module &M,
     return &FAM.getResult<BlockFrequencyAnalysis>(F);
   };
 
-  auto *PSI = &MAM.getResult<ProfileSummaryAnalysis>(M);
+  auto AbandonAndRefreshPSI = [&MAM](Module &M) {
```

Comment:
why we do this through a lambda instead of just invalidate and get result for PSI right here? I think it's cleaner to just do it here and pass in PSI..

Reply:
I like that. We'd need the MAM passed through, but that then also allows dropping the `Lookup`xyz parameters, so simpler API.

Reply:
Oh, I misread part of that. So the reason we can't just fetch the PSI here is because we need the data saved in the module (line 1993) to actually materialize it. I think the rest holds: we can just fetch it when needed from the MAM, just like the other analyses, and simplify the API.